AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,432 @@
+                          status: .published)
+
+        var updatedPost = newPost
+        updatedPost.rating = 5
+        updatedPost.title = "MyUpdatedPost"
+        updatedPost.content = "This is my updated post."
+
+        let saveSyncReceived = expectation(description: "Received create mutation event on subscription for Post")
+        let updateSyncReceived = expectation(description: "Received update mutation event on subscription for Post")
+        // this needs to be commented out once the bug is fixed
+        // currently the API request for delete mutation fails with error message "Conflict resolver rejects mutation."
+        // because of version not being included, so the hub event is never fired
+        // let deleteSyncReceived = expectation(description: "Received delete mutation event on subscription for Post")
+
+        let hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent else {
+                XCTFail("Can't cast payload as mutation event")
+                return
+            }
+
+            guard let post = try? mutationEvent.decodeModel() as? Post, post.id == newPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post, newPost)
+                saveSyncReceived.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.update.rawValue {
+                // this should be removed once the bug is fixed
+                XCTAssertEqual(post, newPost)
+
+                // expected behavior which is currently failing
+                // XCTAssertEqual(post, updatedPost)
+                updateSyncReceived.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.delete.rawValue {
+                XCTAssertEqual(post, updatedPost)
+                // this needs to be commented out once the bug is fixed
+                // deleteSyncReceived.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post, update and delete immediately
+        let saveThenUpdateAndImmediatelyDelete =
+            expectation(description: "Post is saved then updated and deleted immediately")
+        Amplify.DataStore.save(newPost) { result in
+            switch result {
+            case .success:
+                Amplify.DataStore.save(updatedPost) { result in
+                    switch result {
+                    case .success:
+                        Amplify.DataStore.delete(updatedPost) { result in
```

Comment:
this isn't the scenario described in the test comment "    /// - When: A Post is saved with sync complete, updated and deleted immediately
".
```
/// A post is saved with sync complete 
Amplify.DataStore.Save(newPost)
   saveCompleted.fulfill()

wait(for[saveCompleted, saveSynced])

// update and delete immedaitely
Amplify.DataStore.save(updatedPost)
   Amplify.DataStore.delete(updatedPost)
       updateAndImmediatelyDelete.fulfill()

wait(for:[updateAndImmediatelyDelete])

// immediately check that the post has been deleted from local store
Amplify.DataStore.query
   assert(nil)

wait(for:[updateSyncReceived])
// this can be uncommmented when the bug is fixed
// wait(for:[deleteSyncRecevied)

Amplify.DataStore.query

Amplify.API.query
```